### PR TITLE
remove LastHeartbeatTime(pingcap#3080)

### PR DESCRIPTION
**Pull request by songjiansuper**
in operator,this value does not seem to be necessary.
At the same time, every sync will cause this value to change,
which seems to trigger more update events

### What problem does this PR solve?

Every sync will cause LastHeartbeatTime value to change,which seems to trigger more update events.
But there is not much need for this value for the operator, so we can remove it
Fix https://github.com/pingcap/tidb-operator/issues/3080
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [x] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Remove `lastHeartbeatTime` field from `status.tikv.stores` because it will cause the TidbCluster CR to be updated during each sync and actually is useless
```

### CODE REVIEW

**Comment by DanielZhangQD** [comment]
@songjiansuper Thanks for the contribution!
Please sign the CLA.
BTW, what's the sync interval now with a TidbCluster that includes PD, TiDB, TiKV, and TiFlash?

File: pkg/controller/tidbcluster_control.go [hunk 1]
```diff
@@ -114,7 +113,6 @@ func deepEqualExceptHeartbeatTime(newStatus *v1alpha1.TidbClusterStatus, oldStat
 
 func sweepHeartbeatTime(stores map[string]v1alpha1.TiKVStore) {
```
Comment (DanielZhangQD) [inline pkg/controller/tidbcluster_control.go:114]: This function and `deepEqualExceptHeartbeatTime` can be removed.

**Review by july2993 — APPROVED** [review]

**Comment by songjiansuper** [comment]
@DanielZhangQD seems like 30 seconds,but it seems that this 30s is not the same as the 30s of the informer. My own informer resync time is set to 60s, and I will still get an update event in 30s.

**Comment by DanielZhangQD** [comment]
> @DanielZhangQD seems like 30 seconds,but it seems that this 30s is not the same as the 30s of the informer. My own informer resync time is set to 60s, and I will still get an update event in 30s.

It should be controlled by the resync period similar to the code [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/dependences.go#L308).
BTW, did you test the case that deploys the old CRD and a TidbCluster, after all the Pods ready, apply the new CRD in this PR, upgrade to the operator built from this PR and everything works well?

**Comment by DanielZhangQD** [comment]
@songjiansuper Please also help check the comment when you're available, thanks!

**Comment by DanielZhangQD** [comment]
@songjiansuper Just a friendly ping.

**Comment by songjiansuper** [comment]
@DanielZhangQD hello,I’ve been a bit busy with the company’s work recently, I may have to verify the above comment for a while, thx

**Comment by DanielZhangQD** [comment]
> @DanielZhangQD hello,I’ve been a bit busy with the company’s work recently, I may have to verify the above comment for a while, thx

OK, thanks a lot!

**Comment by DanielZhangQD** [comment]
> @DanielZhangQD hello,I’ve been a bit busy with the company’s work recently, I may have to verify the above comment for a while, thx

OK, thanks a lot!

File: pkg/manager/member/tidb_init_manager.go [hunk 2]
```diff
@@ -64,18 +64,7 @@ func NewTiDBInitManager(deps *controller.Dependencies) InitManager {
...
-	tc, err := m.deps.TiDBClusterLister.TidbClusters(ns).Get(tcName)
-	if err != nil {
-		return fmt.Errorf("TidbInitManager.Sync: failed to get tidbcluster %s for TidbInitializer %s/%s, error: %s", tcName, ns, ti.Name, err)
-	}
-	if tc.Spec.TiDB == nil {
-		klog.Infof("TidbInitManager.Sync: Spec.TiDB is nil in tidbcluster %s, skip syncing TidbInitializer %s/%s", tcName, ns, ti.Name)
-		return nil
-	}
-
-	err = m.syncTiDBInitConfigMap(ti)
```
Comment (DanielZhangQD) [inline pkg/manager/member/tidb_init_manager.go:78]: Why remove this block?

Comment (songjiansuper) [reply pkg/manager/member/tidb_init_manager.go:78]: Shocked, I have no impression of it...this doesn't look like it should be deleted.Modify it together with the content to be removed above

File: pkg/manager/member/tidb_init_manager.go [hunk 3]
```diff
@@ -74,7 +74,6 @@ func (m *tidbInitManager) Sync(ti *v1alpha1.TidbInitializer) error {
 		klog.Infof("TidbInitManager.Sync: Spec.TiDB is nil in tidbcluster %s, skip syncing TidbInitializer %s/%s", tcName, ns, ti.Name)
 		return nil
 	}
```
Comment (DanielZhangQD) [inline pkg/manager/member/tidb_init_manager.go:76]:
```suggestion
	}

```

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**